### PR TITLE
solves Gh 30 - Include CSRF token in webapp’s API requests

### PR DESCRIPTION
**Pull request by maisnamrajusingh**
#### Summary
Includes CSRF token in webapp’s API requests
#### Ticket Link
https://github.com/mattermost/mattermost-plugin-webex/issues/30

### CODE REVIEW

File: webapp/src/index.js [hunk 1]
```diff
@@ -20,7 +23,23 @@ class Plugin {
...
+    if (config && config.SiteURL) {
+        basePath = new URL(config.SiteURL).pathname;
+
+        if (basePath && basePath[basePath.length - 1] === '/') {
+            basePath = basePath.substr(0, basePath.length - 1);
+        }
+    }
+
+    return basePath;
+};
```
Comment (hanzei) [inline webapp/src/index.js:45]: Seems like an unrelated change from https://github.com/mattermost/mattermost-plugin-webex/pull/63

Comment (maisnamraju) [reply webapp/src/index.js:45]: looks like the commits got mixed up on this one. Will update for review

**Comment by jasonblais** [comment]
@maisnamrajusingh let us know once this is ready for a review

**Comment by maisnamrajusingh** [comment]
@hanzei  this is ready for review again.

File: webapp/src/client/client.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
 // See License for license information.
 
-import request from 'superagent';
+// import request from 'superagent';
```
Comment (mickmister) [inline webapp/src/client/client.js:4]: This line can be removed

File: webapp/src/client/client.js [hunk 3]
```diff
@@ -15,15 +17,26 @@ export default class Client {
...
+            headers,
+        };
 
-        const response = await request.
-            post(url).
-            send(body).
-            set(headers).
-            type('application/json').
-            accept('application/json');
+        options.headers['X-Requested-With'] = 'XMLHttpRequest';
```
Comment (mickmister) [inline webapp/src/client/client.js:26]: This line can be removed, because `getOptions` already adds this header https://github.com/mattermost/mattermost-redux/blob/3d1028034d7677adfda58e91b9a5dcaf1bc0ff99/src/client/client4.ts#L444

File: webapp/src/index.js [hunk 4]
```diff
@@ -23,4 +22,4 @@ class Plugin {
     }
 }
 
-window.registerPlugin(pluginId, new Plugin());
+window.registerPlugin(pluginId, new Plugin());
```
Comment (mickmister) [inline webapp/src/index.js:25]: Can you please revert changes here so this file is not changed in this PR?

**Review by mickmister — CHANGES_REQUESTED** [review]
Great work @maisnamrajusingh! The logic looks good, I just have some requests to clean up the PR a bit

**Review by cpoile — APPROVED** [review]
Thanks @maisnamrajusingh!

File: webapp/src/client/client.js [hunk 5]
```diff
@@ -15,15 +16,24 @@ export default class Client {
...
+        const text = await response.text();
+
+        throw new ClientError(Client4.url, {
+            message: text || '',
+            status_code: response.status,
+            url,
+        });
     }
-}
+}
```
Comment (mickmister) [inline webapp/src/client/client.js:39]: Looks like there needs to be a newline at the end of this file

File: webapp/src/index.js (hunk 4 above)
Comment (mickmister) [reply webapp/src/index.js:25]: There are still two changes in this file that should be removed to simplify the diff in this file.

- A newline should be added back to the end of this file
- Line 7 should be added back

**Review by mickmister — APPROVED** [review]
Thanks for addressing the feedback @maisnamraju! There are some remaining items but I will approve. Please take a look at the suggestions below.

**Comment by jasonblais** [comment]
@mickmister is this ready to merge? Looks like this PR has two approvals

**Comment by maisnamrajusingh** [comment]
@mickmister Looks like master has been updated with new changes. Should I pull the latest and try to push again ? What is the workflow in such a case

File: webapp/src/index.js [hunk 6]
```diff
@@ -3,8 +3,6 @@
 
 import React from 'react';
 
-import {getConfig} from 'mattermost-redux/selectors/entities/general';
-
```
Comment (mickmister) [inline webapp/src/index.js:7]: Can we add this line back into this file? That should be the last change needed to merge the PR

**Review by mickmister — CHANGES_REQUESTED** [review]
@maisnamrajusingh Thanks for addressing the feedback

There is an import error in `webapp/src/index.js` that is causing issues. The PR looks good otherwise :+1:

Also going forward, please try not to force-push commits after PR review has started, because it makes it difficult to review the history of the PR reviews. Since there's no way to know exactly what has changed between reviews, the entire PR needs to be re-reviewed when commits are force-pushed. It's not a big deal since this PR is fairly small, but for larger PRs it causes the review process to take much longer.

> @mickmister Looks like master has been updated with new changes. Should I pull the latest and try to push again ? What is the workflow in such a case

The state of the PR in relation to master seems fine :+1:

File: webapp/src/index.js (hunk 6 above)
Comment (maisnamrajusingh) [reply webapp/src/index.js:7]: sorry about the force push, had an issue with a commit that shouldn't have gone there. Will keep that in mind.

**Review by maisnamrajusingh — COMMENTED** [review]
ci fialed for some reason. will check and push again.

**Review by mickmister — APPROVED** [review]
@maisnamrajusingh LGTM :+1: Nice work

File: webapp/src/index.js (hunk 6 above)
Comment (mickmister) [reply webapp/src/index.js]: Thank you 🙂

btw having extra/unwanted commits in the PR is fine, because the squash merge will combine it into one commit in the master commit history
